File: py/tux/serve_main.py
"""Internal entry point for the detached server child process."""
from __future__ import annotations

import sys
from pathlib import Path

from .server import start_server


def _arg(name: str):
    argv = sys.argv
    return argv[argv.index(name) + 1] if name in argv else None


def main() -> None:
    cwd = _arg("--cwd") or str(Path.cwd())
    mode = _arg("--mode") or "design"
    host = _arg("--host") or "127.0.0.1"
    port = int(_arg("--port") or 4173)
    target = _arg("--target")
    session = _arg("--session") or "default"
    environment = _arg("--environment") or ("design" if mode == "design" else "development")
    store = _arg("--store") or ".tux/feedback.json"
    project_id = _arg("--project-id") or "default"

    import os
    config = {
        "project_id": project_id,
        "review": {"enabled": True, "store": store, "host": host, "port": port},
        "identity": {
            "provider": "local",
            "user_id": os.environ.get("TUX_USER_ID") or "anonymous",
            "display_name": os.environ.get("TUX_DISPLAY_NAME") or "Anonymous",
            "admins": [],
        },
    }
    server = start_server(
        mode=mode, host=host, port=port, root=_arg("--root"), target=target,
        cwd=cwd, config=config, session=session, environment=environment,
    )
    server.serve_forever()


if __name__ == "__main__":
    main()

File: js/src/serve-main.js
/**
 * Internal entry point for the detached server child process.
 * Not part of the public CLI surface.
 */
import { resolve } from 'node:path';
import { startServer } from './server.js';

function argValue(name) {
  const i = process.argv.indexOf(name);
  return i === -1 ? undefined : process.argv[i + 1];
}

const cwd = resolve(argValue('--cwd') ?? process.cwd());
const mode = argValue('--mode') ?? 'design';
const host = argValue('--host') ?? '127.0.0.1';
const port = Number(argValue('--port') ?? 4173);
const target = argValue('--target') ?? null;
const session = argValue('--session') ?? 'default';
const environment = argValue('--environment') ?? (mode === 'design' ? 'design' : 'development');
const store = argValue('--store') ?? '.tux/feedback.json';
const projectId = argValue('--project-id');
const root = argValue('--root') ?? null;

const server = startServer({
  mode,
  host,
  port,
  root,
  target,
  cwd,
  session,
  environment,
  config: {
    project_id: projectId ?? 'default',
    review: { enabled: true, store, host, port },
    identity: { provider: 'local', user_id: process.env.TUX_USER_ID ?? 'anonymous', display_name: process.env.TUX_DISPLAY_NAME ?? 'Anonymous', admins: [] },
  },
});

server.listen(port, host, () => {
  process.stdout.write(`tux server listening on http://${host}:${port}\n`);
});
